groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -0,0 +1,3338 @@
+
+    template <class TYPE>
+    variant(TYPE&&, BSLSTL_VARIANT_DECLARE_IF_CONSTRUCTS_FROM(variant, TYPE));
+        // Create variant object with an alternative having the value of
+        // specified 'TYPE' object.  The alternative selected is the one for
+        // which expression 'ALT_TYPE x[] = {std::forward<TYPE>(t)};' is a best
+        // match.  In C++03, 'TYPE' is required to be one of the alternative
+        // types. If variant is allocator aware, use the currently installed
+        // default allocator to supply memory.
+
+    template <class TYPE, class... ARGS>
+    explicit variant(bsl::in_place_type_t<TYPE>, ARGS&&...);
+        // Create variant object with an alternative 'TYPE' created in place
+        // using the specified 'args'.  If variant is allocator aware, use the
+        // currently installed default allocator to supply memory.
+
+    template <class TYPE, class INIT_LIST_TYPE, class... ARGS>
+    explicit variant(bsl::in_place_type_t<TYPE>,
+                     std::initializer_list<INIT_LIST_TYPE>,
+                     ARGS&&...);
+        // Create variant object with an alternative 'TYPE' created in place
+        // using the specified 'il' and 'args'.  If variant is allocator aware,
+        // use the currently installed default allocator to supply memory.
+
+    template <size_t INDEX, class... ARGS>
+    explicit variant(bsl::in_place_index_t<INDEX>, ARGS&&...);
+        // Create variant object with an alternative with index 'INDEX' created
+        // in place using the specified 'args'.  If variant is allocator aware,
+        // use the currently installed default allocator to supply memory.
+
+    template <size_t INDEX, class INIT_LIST_TYPE, class... ARGS>
+    explicit variant(bsl::in_place_index_t<INDEX>,
+                     std::initializer_list<INIT_LIST_TYPE>,
+                     ARGS&&...);
+        // Create variant object with an alternative with index 'INDEX' created
+        // in place using the specified 'il' and 'args'.  If variant is
+        // allocator aware, use the currently installed default allocator to
+        // supply memory.
+
+    //  allocator extended constructors
+    variant(bsl::allocator_arg_t, allocator_type allocator);
+        // Create variant object containing the 0th alternative.  If variant is
+        // allocator aware, use the specified allocator to supply memory.
+
+    variant(bsl::allocator_arg_t, allocator_type allocator, const variant&);
+    variant(bsl::allocator_arg_t, allocator_type allocator, variant&&);
+        // Create variant object containing the same alternative as the
+        // 'original' object.  If 'variant' is allocator aware, use the
+        // specified allocator to supply memory.  If the 'original' object is
+        // not 'valueless_by_exception', the contained alternative object is
+        // created by copy construction.  The current copy/move construction
+        // implementation is not 'constexpr'.
```

Comment:
Trying to remember how i got to this wording. I *think* i wanted to distinguish between invoking a direct copy/move construction and construction using an allocator, so the term copy construction was meant to cover both copy and move. The second sentence makes it confusing what I mean, and I haven't actually made a distinction at all. 
Not only that, I have since realised that copy/move construction from a variant can be delegated to the copy/move of the type as the allocator propagation will be handled by the type itself.

This needs two non trivial changes
1) changing the copy/move construction from a variant to not use the allocator explicitly, but to delegate to the type
2) updating the documentation to talk about the way the alternative is constructed (i.e. whether allocator extended construction is done or whether the plain(?) copy/move constructor is invoked)

Reply:
I decided against delegating to the type and going around the `ConstructUtil` facility. I'm not convinced that will make much difference, but it may add unnecessary complexity. Improved the comments